### PR TITLE
Use system format for subscription price display

### PR DESCRIPTION
**Pull request by Hengyu**
We can use `Product.SubscriptionPeriod`'s `formatted` method to display price.

https://developer.apple.com/documentation/storekit/product/subscriptionperiod/4044346-formatted

This PR also simplifies code structure/

### CODE REVIEW

File: Sources/StoreHelper/Core/PrePurchaseSubscriptionInfo.swift [hunk 1]
```diff
@@ -22,30 +22,59 @@ public struct PurchasePriceForDisplay: Hashable, Identifiable {
...
+        product.displayName
+    }
+
+    /// A localized string representation of `price`.
+    public var purchasePrice: String {
+        product.displayPrice
+    }
+
+    /// The price text for display (e.g. "$0.99 / 1 month", "$1.99 / 2 months").
+    public var displayPrice: String {
```
Comment (Hengyu) [inline Sources/StoreHelper/Core/PrePurchaseSubscriptionInfo.swift:59]: Use Apple's recommended API for price display. The API can handle localization as well as singluar and plural units.

File: Sources/StoreHelper/Core/PrePurchaseSubscriptionInfo.swift [hunk 2]
```diff
@@ -56,21 +85,18 @@ public struct PrePurchaseSubscriptionInfo: Hashable {
     /// single element with the promo id (which will be nil) and price/period for the offer.
     /// - If there are no offers, the value of this property will be an array with a single element with the standard price/period.
     public var purchasePriceForDisplay: [PurchasePriceForDisplay]? {
-        if promotionalOffersEligible, let promotionalOffers, promotionalOffers.count > 0 {
+        if promotionalOffersEligible, !promotionalOffers.isEmpty {
```
Comment (Hengyu) [inline Sources/StoreHelper/Core/PrePurchaseSubscriptionInfo.swift:88]: Using `isEmpty` is better than using `count` to tell if an array has elements

File: Sources/StoreHelper/ViewModels/PriceViewModel.swift [hunk 3]
```diff
@@ -70,76 +71,97 @@ public struct PriceViewModel {
...
+    }
     
     private func paymentModeDisplay(mode: Product.SubscriptionOffer.PaymentMode) -> String {
-        switch mode {
-            case .freeTrial:    return "Free trial"
-            case .payAsYouGo:   return "Pay as you go"
-            case .payUpFront:   return "Pay up front"
-            default:            return "Unknown"
+        if #available(iOS 15.4, macOS 12.3, tvOS 15.4, watchOS 8.5, visionOS 1.0, *) {
+            mode.localizedDescription
```
Comment (Hengyu) [inline Sources/StoreHelper/ViewModels/PriceViewModel.swift:157]: use Apple's API to support for localization

**Comment by Hengyu** [comment]
@russell-archer PTAL

File: Sources/StoreHelper/Core/PrePurchaseSubscriptionInfo.swift (hunk 1 above)
Comment (russell-archer) [reply Sources/StoreHelper/Core/PrePurchaseSubscriptionInfo.swift:59]: Very good idea. This wasn't available when I first wrote the code so I had to create my display logic. Much better to use Apple's that supports proper localization.

File: Sources/StoreHelper/Core/PrePurchaseSubscriptionInfo.swift (hunk 2 above)
Comment (russell-archer) [reply Sources/StoreHelper/Core/PrePurchaseSubscriptionInfo.swift:88]: I checked and yes you're correct! Apple recommend using isEmpty!

**Comment by russell-archer** [comment]
Really excellent PR! Thank you Hengyu!

**Pull request merged**